### PR TITLE
Fixes #2585 - add test to DeleteQualifier in pywbem_mock

### LINKED ISSUE #2585: Check and reject DeleteQualifier if the qualifier is used
A well behaved WBEM server should reject the DeleteQualifier operation if the qualifier is used anywhere.

The pywbem mock support currently just deletes the qualifier regardless of whether it is being used.

### PR DESCRIPTION
**Pull request by KSchopmeyer**
See commit message for details

### CODE REVIEW

File: pywbem_mock/_mainprovider.py [hunk 1]
```diff
@@ -1171,8 +1171,35 @@ def DeleteQualifier(self, namespace, QualifierName):
             :exc:`~pywbem.CIMError`: (CIM_ERR_INVALID_NAMESPACE)
             :exc:`~pywbem.CIMError`: (CIM_ERR_INVALID_PARAMETER)
             :exc:`~pywbem.CIMError`: (CIM_ERR_NOT_FOUND)
-        """
-
+            :exc:`~pywbem.CIMError`: (CIM_ERR_FAILED)
+
+        """
+        def qualifier_exists(cls, qualifier_name):
```
Comment (andy-maier) [inline pywbem_mock/_mainprovider.py:1177]: I would prefer a terminology that the qualifier is used on the class component. Existence is something I would attribute to the qualifier declaration.

File: pywbem_mock/_mainprovider.py [hunk 2]
```diff
@@ -1171,8 +1171,35 @@ def DeleteQualifier(self, namespace, QualifierName):
...
+            """
+            Test for qualifier exists in any component of the class cls.
+
+            Parameters:
+              cls ()
```
Comment (andy-maier) [inline pywbem_mock/_mainprovider.py:1182]: Add the type of the cls parameter and a colon at the end:
`cls (CIMClass):`

**Review by andy-maier — CHANGES_REQUESTED** [review]
Minor comments

File: pywbem_mock/_mainprovider.py (hunk 1 above)
Comment (KSchopmeyer) [reply pywbem_mock/_mainprovider.py:1177]: qualifier_exiists_in_cls. DONE

File: pywbem_mock/_mainprovider.py (hunk 2 above)
Comment (KSchopmeyer) [reply pywbem_mock/_mainprovider.py:1182]: DONE

**Comment by KSchopmeyer** [comment]
Made fixes.  Now includes test to assure that reset does not disable statistics.  Ready for review of last changes

File: pywbem_mock/_mainprovider.py (hunk 1 above)
Comment (andy-maier) [reply pywbem_mock/_mainprovider.py]: Not what I had in mind, but I see you like the "exists" terminology. Ok with me.

**Pull request merged**